app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -40,17 +42,26 @@ echo "#"
 
 <% if plugin_present?('katello') -%>
 if [ x$ID = xrhel ] || [ x$ID = xcentos ]; then
-    CONSUMER_RPM=$(mktemp --suffix .rpm)
+    register_katello_host(){
+        UUID=$(subscription-manager identity | head -1 | awk '{print $3}')
+        curl --request POST "<%= foreman_server_url %>/register?uuid=$UUID" \
+            <%= headers.join(' ') %> \
+            -d '{ "host": { "organization_id": "<%= @organization.id -%>"
+            <%= ", \"location_id\": \"#{@location.id}\"" if @location -%>
+            <%= ", \"hostgroup_id\": \"#{@hostgroup.id}\"" if @hostgroup -%>
```

Comment:
note for me: `operatingsystem_id` here is not passed, since we get it from sub-man information